src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+
+        /// <summary>
+        /// Authorization client
+        /// </summary>
+        private IAuthorizationManagementClient authorizationManagementClient;
+
+        public IAuthorizationManagementClient AuthorizationManagementClient
+        {
+            get
+            {
+                return authorizationManagementClient = authorizationManagementClient ?? AzureSession.Instance.ClientFactory.CreateArmClient<AuthorizationManagementClient>(DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+ 
+            }
+            set => authorizationManagementClient = value;
+        }
+
+        /// <summary>
+        /// ARM client
+        /// </summary>
+        private IResourceManagementClient resourceManagerClient;
+        public IResourceManagementClient ResourceManagerClient
+        { 
+            get
+            {
+                return resourceManagerClient = resourceManagerClient ?? new ResourceManagementClient(
+                                                   DefaultProfile.DefaultContext.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
+                                                   ClientCredentials);
+            }
+            set => resourceManagerClient = value;
+        }
+
+        #endregion Properties
+
+        #region Cmdlet Overrides
+        protected override void WriteExceptionError(Exception ex)
+        {
+            var aggEx = ex as AggregateException;
+
+            if (aggEx != null && aggEx.InnerExceptions != null)
+            {
+                foreach (var e in aggEx.Flatten().InnerExceptions)
+                {
+                    WriteExceptionError(e);
+                }
+
+                return;
+            }
+
+            throw ex;
+        }
+        #endregion Cmdlet Overrides
+
+        #region Protected Methods
+        /// <summary>
+        /// Check if an assignment with the same name exists
+        /// </summary>
+        /// <param name="scope"> Subscription scope</param>
+        /// <param name="name"> Assignment name</param>
+        /// <param name="invertCheck"> Inverts the check logic from check for exists to check for not-exits </param>
+        protected void CheckIfAssignmentExist(string scope, string name, bool invertCheck)
+        {
+            PSBlueprintAssignment existingAssignment = null;
+
+            try
+            {
+                existingAssignment = BlueprintClient.GetBlueprintAssignment(scope, name);
+            }
+            catch (Exception ex)
+            {
+                if (ex is CloudException cex && cex.Response.StatusCode != System.Net.HttpStatusCode.NotFound)
+                {
+                    throw new CloudException(cex.Message);
+                }
+            }
+
+            if (invertCheck)
+            {
+                if (existingAssignment == null)
+                {
+                    throw new Exception(string.Format(Resources.AssignmentNotExist, name, scope));
```

Comment:
>string.Format( [](start = 40, length = 14)

All string.Format calls should specify a culture, so any numbers get formatted correctly.

Reply:
Ones that deal with localized strings should probably use either CurrentCulture or CurrentUICulture (not sure which). Anything that is not localized text (like scope strings, for example) should use InvariantCulture

---
In reply to: [257093111](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257093111) [](ancestors = 257093111)

Reply:
Also we probably should turn on some static analysis to catch stuff like this for us.

---
In reply to: [257118910](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257118910) [](ancestors = 257118910,257093111)

Reply:
I'll follow up on static analysis offline. Resolving rest of the comments as by design. 

---
In reply to: [257119216](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257119216) [](ancestors = 257119216,257118910,257093111)